File: 2_7/2_7.cs

using System;
using System.Collections.Generic;

public class Program
{
    public static string FindDestination(List<KeyValuePair<string, string>> destinations)
    {
        Dictionary<string, string> map = new Dictionary<string, string>();
        foreach (var p in destinations)
        {
            if (p.Key == p.Value)
            {
                return "обратитесь к специалисту";
            }
            map[p.Key] = p.Value;
        }

        string current = destinations[0].Key;
        while (map.ContainsKey(current))
        {
            current = map[current];
        }
        return current;
    }

    public static void Main()
    {
        List<KeyValuePair<string, string>> destinations = new List<KeyValuePair<string, string>>
        {
            new KeyValuePair<string, string>("London", "Istanbul"),
            new KeyValuePair<string, string>("Novosibirsk", "London"),
            new KeyValuePair<string, string>("Istanbul", "Beijing")
        };
        Console.WriteLine("Output: " + FindDestination(destinations));
    }
}

File: 2_7/2_7.js

function findDestination(destinations) {
    const map = new Map();
    for (const [from, to] of destinations) {
      if (from === to) {
        return "обратитесь к специалисту";
      }
      map.set(from, to);
    }
  
    let current = destinations[0][0];
    while (map.has(current)) {
      current = map.get(current);
    }
    return current;
  }
  
  const destinations = [
    ["Новосибирск", "Дубай"],
    ["Новосибирск", "Искитим"],
    ["Искитим", "Линево"]
  ];
  console.log("Output:", findDestination(destinations));
  
  